tests/model_fields/tests.py

```diff
@@ -183,6 +183,21 @@ def test_choices(self):
             self.choices_from_callable.choices.func(), [(0, "0"), (1, "1"), (2, "2")]
         )
 
+    def test_choices_slice(self):
+        self.assertEqual(self.empty_choices.choices[:2], [])
+        self.assertEqual(self.empty_choices_bool.choices[:2], [])
+        self.assertEqual(self.empty_choices_text.choices[:2], [])
+        self.assertEqual(self.with_choices.choices[:2], [(1, "A")])
+        self.assertEqual(self.with_choices_dict.choices[:2], [(1, "A")])
+        self.assertEqual(
+            self.with_choices_nested_dict.choices[:2], [("Thing", [(1, "A")])]
+        )
+        self.assertEqual(self.choices_from_iterator.choices[:2], [(0, "0"), (1, "1")])
+        self.assertEqual(
+            self.choices_from_callable.choices.func()[:2], [(0, "0"), (1, "1")]
+        )
+        self.assertEqual(self.choices_from_callable.choices[:2], [(0, "0"), (1, "1")])
+
```

Comment:
Shall we consider adding a few test cases for negative slices?
```suggestion
        self.assertEqual(self.choices_from_callable.choices[-1:-2], [])
        self.assertEqual(self.choices_from_callable.choices[-2:-1], [(1, "1")])

```